pkg/kubelet/kubelet.go

```diff
@@ -2535,6 +2525,22 @@ func (kl *Kubelet) HandlePodAdditions(pods []*v1.Pod) {
 	}
 }
 
+// updateContainerResourceAllocation updates AllocatedResources values
+// (for cpu & memory) from checkpoint store
+func (kl *Kubelet) updateContainerResourceAllocation(pod *v1.Pod) {
+	for _, c := range pod.Spec.Containers {
+		allocatedResources, found := kl.statusManager.GetContainerResourceAllocation(string(pod.UID), c.Name)
+		if c.Resources.Requests != nil && found {
+			if _, ok := allocatedResources[v1.ResourceCPU]; ok {
+				c.Resources.Requests[v1.ResourceCPU] = allocatedResources[v1.ResourceCPU]
+			}
+			if _, ok := allocatedResources[v1.ResourceMemory]; ok {
+				c.Resources.Requests[v1.ResourceMemory] = allocatedResources[v1.ResourceMemory]
+			}
```

Comment:
Would it make sense to set requests values from checkpoint so we don't have different data in the checkpoint and in the c.Resources.Requests? Alternatively we can save/restore data to/from checkpoint in a bit smarter way so that if memory or cpu is not requested it will not be saved and restored?

Not 100% sure it makes sense, but this difference between checkpoint and memory data is a bit concerning.

@vinaykul @SergeyKanzhelev @aheng-ch WDYT?

Reply:
@bart0sh GetContainerResourceAllocation reads from checkpoint **because** the Requests and checkpoint values (what kubelet agreed to at a prior point) can differ. Did I miss something?

Reply:
sorry, this PR somehow slipped off my radar. I was thinking that some fields with `nil` value can be converted to empty lists before checkpointing them, so when they're read from a checkpoint we don't need to check if they're nil.